tests/mir-opt/dead-store-elimination/call_arg_copy.move_simple.DeadStoreElimination.panic-unwind.diff

```diff
@@ -0,0 +1,23 @@
+- // MIR for `move_simple` before DeadStoreElimination
++ // MIR for `move_simple` after DeadStoreElimination
+  
+  fn move_simple(_1: i32) -> () {
+      debug x => _1;
+      let mut _0: ();
+      let _2: ();
+-     let mut _3: i32;
+-     let mut _4: i32;
+  
+      bb0: {
+          StorageLive(_2);
+-         _2 = use_both(_1, _1) -> [return: bb1, unwind continue];
++         _2 = use_both(_1, move _1) -> [return: bb1, unwind continue];
```

Comment:
@RalfJung IMO, the thorny question is here. It this transformation ok?

Reply:
That's a very good question. It is okay under the `move` semantics I implemented in Miri recently (assuming `_1` is not read from again later), because there we do left-to-right evaluation so the first `_1` is copied before the `move` action. But I don't know if there's a risk of codegen backends doing things that would cause problems here. Also `move` semantics are far from clear; I don't think we already have the semantics we want here -- we just have something that justifies what codegen does.

I think, similar to how we avoid making assumptions about the aliasing model, we should be conservative and avoid this if any of the other arguments (or the return value) could alias the newly `move`d argument.

Cc @bjorn3

Reply:
Relevant UCG issue: https://github.com/rust-lang/unsafe-code-guidelines/issues/416

Reply:
I think this is fine given that the first `_1` is copied before the call passes the address of the second argument. It may even be ok to do `use_both(move _1, _1)` with current backends, but I don't think whatever semantics we give to `move` arguments should allow that as semantically the move of the first arg happens before the copy of the second arg.